src/FetchInterceptor.js

```diff
@@ -0,0 +1,329 @@
+    this.handleResponse = this.handleResponse.bind(this);
+  }
+
+  /**
+   * Configures fetch interceptor with given config object. All required properties can optionally
+   * return a promise which will be resolved by fetch interceptor automatically.
+   *
+   * @param config
+   *
+   * (Required) Prepare fetch request for renewing new access token
+   *   createAccessTokenRequest: (refreshToken) => request,
+   *
+   * (Required) Parses access token from access token response
+   *   parseAccessToken: (response) => accessToken,
+   *
+   * (Required) Defines whether interceptor will intercept this request or just let it pass through
+   *   shouldIntercept: (request) => boolean,
+   *
+   * (Required) Defines whether access token will be invalidated after this response
+   *   shouldInvalidateAccessToken: (response) => boolean,
+   *
+   * (Required) Adds authorization for intercepted requests
+   *   authorizeRequest: (request) => authorizedRequest,
+   *
+   * Number of retries after initial request was unauthorized
+   *   fetchRetryCount: 1,
+   *
+   * Event invoked when access token has changed
+   *   onAccessTokenChange: null,
+   *
+   * Event invoked when response is resolved
+   *   onResponse: null,
+   *
+   */
+  configure(config) {
+    this.config = { ...this.config, ...config };
+
+    if (!this.isConfigValid(this.config)) {
+      throw new Error(ERROR_INVALID_CONFIG);
+    }
+
+    this.accessTokenProvider = new AccessTokenProvider(this.fetch, this.config);
+  }
+
+  /**
+   * Authorizes fetch interceptor with given renew token
+   * @param refreshToken
+   * @param accessToken
+   */
+  authorize(refreshToken, accessToken) {
+    this.accessTokenProvider.authorize(refreshToken, accessToken);
+  }
+
+  /**
+   * Returns current authorization for fetch fetchInterceptor
+   * @returns {{accessToken: string, refreshToken: string}}
+   */
+  getAuthorization() {
+    return this.accessTokenProvider.getAuthorization();
+  }
+
+  /**
+   * Clears authorization tokens. Call this to effectively log out user from fetch interceptor.
+   */
+  clear() {
+    this.accessTokenProvider.clear();
+  }
+
+  /**
+   * Main intercept method, you should chain this inside wrapped fetch call
+   * @param args Args initially provided to fetch method
+   * @returns {Promise} Promise which resolves the same way as fetch would
+   */
+  intercept(...args) {
+    return new Promise((resolve, reject) => this.resolveIntercept(resolve, reject, ...args));
+  }
+
+  isConfigValid() {
+    return this.config.shouldIntercept &&
+      this.config.authorizeRequest &&
+      this.config.createAccessTokenRequest &&
+      this.config.parseAccessToken;
+  }
+
+  resolveIntercept(resolve, reject, ...args) {
+    const request = new Request(...args);
+    const { accessToken } = this.accessTokenProvider.getAuthorization();
+    const requestUnit = this.createRequestUnit(request, resolve, reject);
```

Comment:
Renamed to RequestContext, also added comments on createRequestContext method describing what it represents.